components/admin_console/schema_admin_settings.jsx

```diff
@@ -621,16 +631,22 @@ export default class SchemaAdminSettings extends React.Component {
         this.handleChange(id, s.replace('+', '-').replace('/', '_'));
     }
 
-    handleChange = (id, value) => {
-        let saveNeeded = 'config';
-        if (this.state.saveNeeded === 'permissions') {
-            saveNeeded = 'both';
-        }
+    handleChange = (id, value, confirm = false, doSubmit = false, warning = false) => {
+        const saveNeeded = this.state.saveNeeded === 'permissions' ? 'config' : 'both';
+        const confirmNeeded = confirm ? id : this.state.confirmNeeded;
```

Comment:
nit: Can this be
```suggestion
        const confirmNeeded = confirm ? confirm : this.state.confirmNeeded;
```
I feel it makes more sense to keep it a `boolean` value rather than make it a `string`. Not crucial but less confusing.

Reply:
I decided to keep it a string in case you have multiple components on one page that require confirmation - this ensures that only the component that requires the confirmation gets the `showConfirm` param sent to it - I guess I could keep it a boolean for now since that case isnt really present at the moment and it may be premature to design for that case

Reply:
@hahmadia 

I added some logic to handle removing the existing `confirmNeeded` if the `id = confirmNeeded` and `confirm = false` 

If `confirmNeeded` is a boolean there is no way to determine which component required the confirmation preventing us from ever knowing when to remove the confirmation dialog

I also renamed `confirmNeeded` to `confirmNeededId` to make it more explicit that it _is_ a setting id and not a boolean